lib/steep/types/tuple.rb

```diff
@@ -0,0 +1,53 @@
+module Steep
+  module Types
+    class Tuple
+      # @implements Steep__Types__Name
+      # @type const TypeName::Interface: Steep__TypeName.class
+      # @type const TypeName::Module: Steep__TypeName.class
+      # @type const TypeName::Instance: Steep__TypeName.class
+
+      # @dynamic types
+      attr_reader :types
+
+      def initialize(types:)
+        @types = types
+      end
+
+      def self.interface(types: [])
```

Comment:
I don't think we have to have these factory methods. (I had implemented these for `Name` type.)